core/capabilities/triggers/logevent/service.go

```diff
@@ -0,0 +1,161 @@
+package logevent
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/core"
+)
+
+const ID = "log-event-trigger-%s-%d@1.0.0"
+
+const defaultSendChannelBufferSize = 1000
+
+// Log Event Trigger Capability Input
+type Input struct {
+}
+
+// Log Event Trigger Capability Payload
+type Payload struct {
+	// Time that Log Event Trigger's task execution occurred (RFC3339Nano formatted)
+	ActualExecutionTime string
+}
+
+// Log Event Trigger Capabilities Manager
+// Manages different log event triggers using an underlying triggerStore
+type TriggerService struct {
+	capabilities.CapabilityInfo
+	capabilities.Validator[RequestConfig, Input, capabilities.TriggerResponse]
+	lggr           logger.Logger
+	triggers       CapabilitiesStore[logEventTrigger, capabilities.TriggerResponse]
+	relayer        core.Relayer
+	logEventConfig Config
+	stopChan       services.StopChan
+	wg             sync.WaitGroup
+}
+
+// Common capability level config across all workflows
+type Config struct {
+	ChainID        uint64 `json:"chainId"`
+	Network        string `json:"network"`
+	LookbackBlocks uint64 `json:"lookbakBlocks"`
+	PollPeriod     uint32 `json:"pollPeriod"`
+}
+
+func (config Config) Version(capabilityVersion string) string {
+	return fmt.Sprintf(capabilityVersion, config.Network, config.ChainID)
+}
+
+type Params struct {
```

Comment:
This separate struct adds a lot of verbosity compared to flat constructor parameters. What is the advantage?

Reply:
This seems to be the convention used in reference capabilities like cron trigger and kv store in capabilities repo.

Reply:
Well we need to present an internally consistent set of identifiers within each package. These look like `logevent.TriggerServiceParams` to me, rather than `logevent.Params`. But if we really need this kind of struct based configuration API, then there are cleaner ways to do it, like with a builder/factory struct that has a method:
```suggestion
type TriggerServiceFactory struct {
```
```go
func (f *TriggerServiceFactory) New() *TriggerService {
```
Or by using an initialization method:
```go
type TriggerService struct {
	Relayer
	LogEventConfig
	// unexported fields
}
func (s *TriggerService) Init(logger.Logger) {
	// normal construction
}
```
Both of these approaches have far less verbose APIs, because you don't need to repeat the package and the type prefix from the caller.

Reply:
Switched to flat constructor parameters for now, for simplicity. We could discuss whether a factory approach is better later.